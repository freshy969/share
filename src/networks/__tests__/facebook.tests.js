import { when } from 'mobx';
import Facebook from '../facebook';

describe('Share > Facebook', () => {
  test('url', () => {
    const entityLinkMock = { entityLink: jest.fn() };

    const facebook = Facebook.named('FacebookMocked')
      .views(() => entityLinkMock)
      .create({});

    const entityLink = 'https://demo.frontity.com/the-beauty-of-gullfoss';
    facebook.entityLink.mockReturnValueOnce(entityLink);

    expect(facebook.url({ type: 'post', id: 60, quote: 'q', hashtag: '#tag' })).toMatchSnapshot();
    expect(facebook.entityLink.mock.calls).toMatchSnapshot();
    expect(facebook.entityLink.mock.results).toMatchSnapshot();
  });

  test('requestCount success', async done => {
    const request = jest.fn().mockReturnValue(
      Promise.resolve({
        share: {
          comment_count: 0,
          share_count: 123,
        },
        id: 'https://demo.frontity.com/the-beauty-of-gullfoss',
      }),
    );

    const entityLinkMock = { entityLink: jest.fn() };
    const facebook = Facebook.named('FacebookMocked')
      .views(() => entityLinkMock)
      .create({}, { request });

    const entityLink = 'https://demo.frontity.com/the-beauty-of-gullfoss';
    facebook.entityLink.mockReturnValueOnce(entityLink);

    when(
      () => facebook.count({ type: 'post', id: 60 }),
      () => {
        expect(facebook.count({ type: 'post', id: 60 })).toBe(123);
        expect(facebook.entityLink.mock.calls).toMatchSnapshot();
        expect(facebook.entityLink.mock.results).toMatchSnapshot();
        expect(request.mock.calls).toMatchSnapshot();
        expect(request.mock.results).toMatchSnapshot();
        done();
      },
    );

    await facebook.requestCount({ type: 'post', id: 60 });
  });

  test('requestCount fails', async () => {
    const request = jest.fn();

    const entityLinkMock = { entityLink: jest.fn() };
    const facebook = Facebook.named('FacebookMocked')
      .views(() => entityLinkMock)
      .create({}, { request });

    const entityLink = 'https://demo.frontity.com/the-beauty-of-gullfoss';
    facebook.entityLink.mockReturnValue(entityLink);

    // First request (request fails)
    request.mockReturnValue(Promise.reject(new Error()));
    await facebook.requestCount({ type: 'post', id: 60 });
    expect(facebook.count({ type: 'post', id: 60 })).toBe(null);

    // Second request (bad response)
    request.mockReturnValue(Promise.resolve({ bad: 'response'}));
    await facebook.requestCount({ type: 'post', id: 60 });
    expect(facebook.count({ type: 'post', id: 60 })).toBe(null);

    expect(request.mock.calls).toMatchSnapshot();
  });
});
