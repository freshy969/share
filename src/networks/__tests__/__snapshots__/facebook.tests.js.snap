// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > Facebook requestCount fails 1`] = `
Array [
  Array [
    "https://graph.facebook.com?id=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauty-of-gullfoss",
  ],
  Array [
    "https://graph.facebook.com?id=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauty-of-gullfoss",
  ],
]
`;

exports[`Share > Facebook requestCount success 1`] = `
Array [
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
]
`;

exports[`Share > Facebook requestCount success 2`] = `
Array [
  Object {
    "isThrow": false,
    "value": "https://demo.frontity.com/the-beauty-of-gullfoss",
  },
]
`;

exports[`Share > Facebook requestCount success 3`] = `
Array [
  Array [
    "https://graph.facebook.com?id=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauty-of-gullfoss",
  ],
]
`;

exports[`Share > Facebook requestCount success 4`] = `
Array [
  Object {
    "isThrow": false,
    "value": Object {
      "accept": [MockFunction] {
        "calls": Array [
          Array [
            "json",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      },
    },
  },
]
`;

exports[`Share > Facebook url 1`] = `"https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauty-of-gullfoss&quote=q&hashtag=%23tag"`;
